### ISSUE TITLE
[Feature Request] Exclude member expressions from validation (CSS modules support)

### ISSUE THREAD

**Issue by bengry** [schoero/eslint-plugin-better-tailwindcss#286]
## Problem

When using CSS modules alongside Tailwind utilities like `cva` or `clsx`, member expressions like `styles['some-class']` or `styles.someClass` are incorrectly flagged as unregistered Tailwind classes.

Example:

```tsx
import styles from "./Component.module.css";
import { cva } from "cva";
import { classes } from '@utils/classes'; // see https://gist.github.com/bengry/a11b9e5f19341ebdf42a42b68b728da4 for more info on this utility

const componentStyles = cva({
  base: classes({
    style: "text-sm rounded-md", // ✅ Tailwind classes - should be validated
    invalid: [styles["text-area"]], // ❌ CSS module - incorrectly flagged as unregistered class "text-area"
  }),
});
```

The plugin sees `'text-area'` as a string literal and tries to validate it as a Tailwind class, but it's actually a CSS module property access key.

## Current workaround

Adding each CSS module class name to the `ignore` list:

```js
'better-tailwindcss/no-unregistered-classes': ['warn', {
  ignore: ['text-area', 'another-css-class', ...],
}],
```

or use `eslint-disable-next-line` for each usage

This doesn't scale well as it requires manually maintaining a list of all CSS module class names, or fill the codebase with `eslint-disable-next-line` comments.

## Proposed solution

Add an option to exclude strings used in member expressions on specific object names:

```js
'better-tailwindcss/no-unregistered-classes': ['warn', {
  excludeMemberExpressions: ['styles'],
}],
```

This would tell the plugin to skip validation for any string used as a property access on variables matching those names:

- `styles['foo-bar']` → skip `'foo-bar'`
- `styles.fooBar` → skip `fooBar`

## Alternatives considered

1. **Using `variables` configuration** - Tried `variables: [['styles', []]]` but this only affects variable declarations, not member expressions used inside function calls.

2. **Regex in `ignore`** - Could use a pattern like `^[a-z]+(-[a-z]+)*$` but this is too broad and would exclude legitimate Tailwind class errors, plus will probably slow down ESLint significantly.

## Additional context

CSS modules are commonly used alongside Tailwind for:

- Complex animations
- Complex CSS that even if possible with Tailwind, would be unreadable (e.g. `grid-template-areas`)
- Pseudo-element styling not available in Tailwind
- Third-party component overrides

A first-class way to exclude CSS module patterns would greatly improve DX for projects using both.

**Comment by schoero** [comment]
I'm not sure if I ever came across a real world use case where indexed access string literals were Tailwind CSS classes.
I think the easiest fix would be if they get ignored in all current matchers.

If there is a real world use case for them, in my opinion it would be cleaner to add a new matcher:

```ts
{
  "callees": [
    [
      "classes", [
        {
          "match": "indexAccessKey"
        }
      ]
    ]
  ]
}
```

What do you think?

**Comment by bengry** [comment]
> I'm not sure if I ever came across a real world use case where indexed access string literals were Tailwind CSS classes. I think the easiest fix would be if they get ignored in all current matchers.
> 
> If there is a real world use case for them, in my opinion it would be cleaner to add a new matcher:
> 
> {
>   "callees": [
>     [
>       "classes", [
>         {
>           "match": "indexAccessKey"
>         }
>       ]
>     ]
>   ]
> }
> What do you think?

LGTM. I think adding this as a default behavior makes sense. Not sure if we need to complicate things with the new matcher. I can't think of a use-case for using them as index access keys, but I might also be missing something.
